### PR TITLE
[HOTFIX] Advanced command to send rosetta transactions

### LINKED ISSUE #8140: Broadcast generated Ledger transactions with command in Mina advanced (it’s a Rosetta.Rendered.t)
Should be very similar to test-transaction but actually broadcast it

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds the `advanced send-rosetta-transactions` command, which reads rosetta-format commands off stdin and sends them to the daemon over GraphQL.

The error messages are currently not great.

**This needs to be tested before merging.**

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them:

Closes #8140

### CODE REVIEW

**Review by bkase — APPROVED** [review]
when it's tested, it looks good!

**Comment by bkase** [comment]
!approved-for-mainnet

File: src/lib/rosetta_lib/transaction.ml [hunk 1]
```diff
@@ -446,3 +446,70 @@ module Signed = struct
...
+            , Option.bind rosetta_transaction_rendered.create_token_account
+                ~f:(fun {memo; _} -> memo) )
+        | `Mint_tokens ->
+            ( Option.bind rosetta_transaction_rendered.mint_tokens
+                ~f:(fun {valid_until; _} -> valid_until)
+            , Option.bind rosetta_transaction_rendered.mint_tokens
+                ~f:(fun {memo; _} -> memo) )
+      in
+      let pk (`Pk x) =
+        Signature_lib.Public_key.Compressed.of_base58_check_exn x
```
Comment (psteckler) [inline src/lib/rosetta_lib/transaction.ml:493]: Should this be an error, if the decoding fails?

Comment (mrmr1993) [reply src/lib/rosetta_lib/transaction.ml:493]: It should be already, it gets caught as one in the `Or_error.try_with_join` above

**Comment by mrmr1993** [comment]
This worked successfully in testing on a local network (with thanks to @jspada):
```
$ _build/default/src/app/cli/src/mina.exe advanced send-rosetta-transaction
{
  "signature": "389ac7d4077f3d485c1494782870979faa222cd906b25b2687333a92f41e40b925adb08705eddf2a7098e5ac9938498e8a0ce7c70b25ea392f4846b854086d43",
  "payment": {
    "to": "B62qnzbXmRNo9q32n4SNu2mpB8e7FYYLH8NmaX6oFCBYjjQ8SbD7uzV",
    "from": "B62qnzbXmRNo9q32n4SNu2mpB8e7FYYLH8NmaX6oFCBYjjQ8SbD7uzV",
    "fee": "10000000",
    "token": "1",
    "nonce": "0",
    "memo": null,
    "amount": "1000000000",
    "valid_until": "4294967295"
  },
  "stake_delegation": null,
  "create_token": null,
  "create_token_account": null,
  "mint_tokens": null
}
Dispatched command with TRANSACTION_ID 4Rs6xMHVyo1J1TTQDCzynYvWPdBr9QRY23fuqvnHFvQiM2B4YD14dtWY2sDccbgx7eh5FYAsPSNrZ2M3AqEPfXoXjNxuWTgiwkL3nwaTaGhSoPA7LcfqiWT6uN9oookDeR6ZSMfd2bs9QSPRit8gPt3FSrDo8i3qM383AEG6g5pEm2i1m1cTTwrUe7y4Z2eB6DdWKhFhoYnx5ndQRtZt3D3o7gojdwwdpRzrWZgT9KcJfbXdZNTfXxr3G1VCVqgqmNJ6iGzn4uTGqpqCmJf6zRn196SWUyZ4DYrphvGF8GhJZYyPTwA2BMTBkF9xWC9zzJP8ZrAeyV1qV8k94dZaiyVt7Fac3r6BcDaMonMcpvnGy8fHF9Q25L2tjzHzwKMePkJztB5r
$ _build/default/src/app/cli/src/mina.exe advanced send-rosetta-transaction
{
  "signature": "37b3940d03b92733fd617cc6e31b01028dc89f7af513b665b6f996f8c01ad1e7349842b8344f503bceb8546eb4bf0a99544bae4f4a17da3f67d91e8b9eaa0df3",
  "payment": {
    "to": "B62qnzbXmRNo9q32n4SNu2mpB8e7FYYLH8NmaX6oFCBYjjQ8SbD7uzV",
    "from": "B62qnzbXmRNo9q32n4SNu2mpB8e7FYYLH8NmaX6oFCBYjjQ8SbD7uzV",
    "fee": "10000000",
    "token": "1",
    "nonce": "1",
    "memo": null,
    "amount": "1000000000",
    "valid_until": "4294967295"
  },
  "stake_delegation": null,
  "create_token": null,
  "create_token_account": null,
  "mint_tokens": null
}
Dispatched command with TRANSACTION_ID 4Rs6xMHVyo1J1TTQDCzynYvWPdBr9QRY23fuqvnHFvQiM2B4YD14dtWY2sDccbgx7eh5FYAsPSNrfgkVQcE4FTx5eh1tNXjALEuxCBxMquykfcfmguvtqxbskirQ1muX2NSoPLbZpxmbZzQ6B6TeN8z1vqBGdutnK4Y16V3SB6EsWN1wRRNgP1PVdBhiRBtf2iJMCKbPNTYh5kRY1CxYSjWmkQkrdqyk2ckthzd8puEme9mydoGb1ay5Q4Kx1ejNL5N8jtzabjGmL9MYFCbgGE94mX1KqEkysn3kW7p4MXq2XcLdkiAe8YaGab3yebX6znUcskFQVKoVzFS8FCZhTjhhNoMCgztTR1nL1UGVTFZ1ey2EjoJPLYK2dQvqDe9x8N91W5mg
```

Fixing up the client sdk compilation, and then this should be ready.

File: src/app/cli/src/init/client.ml [hunk 2]
```diff
@@ -770,23 +770,30 @@ let send_rosetta_transactions_graphql =
...
+                         graphql_endpoint
+                     in
+                     let (`UserCommand user_command) =
+                       (response#sendRosettaTransaction)#userCommand
+                     in
+                     printf "Dispatched command with TRANSACTION_ID %s\n"
+                       user_command#id )
+                 jsons ;
+               don't_wait_for !prev ;
+               !prev )
```
Comment (bkase) [inline src/app/cli/src/init/client.ml:796]: seems better than a block_on_async (and if you tested it and it works, then 👍 )

**Review by bkase — APPROVED** [review]

**Comment by mrmr1993** [comment]
CI is happy. Merging.

**Comment by jspada** [comment]
Worked for me after learning I needed Ctrl-d to trigger sending.

**Pull request merged**